src/uu/nl/src/nl.rs

```diff
@@ -106,6 +106,42 @@ impl NumberFormat {
     }
 }
 
+enum SectionDelimiter {
+    Header,
+    Body,
+    Footer,
+    None,
+}
+
+struct SectionDelimiterParser {
+    delimiter: String,
+}
+
+impl SectionDelimiterParser {
+    fn new(delimiter: &str) -> Self {
+        let delimiter = if delimiter.chars().count() == 1 {
+            format!("{delimiter}:")
+        } else {
+            delimiter.to_owned()
+        };
+
+        Self { delimiter }
+    }
+
+    fn parse(&self, s: &str) -> SectionDelimiter {
+        if self.delimiter.is_empty() {
+            return SectionDelimiter::None;
+        }
+
+        match s {
+            _ if s == self.delimiter.repeat(3) => SectionDelimiter::Header,
+            _ if s == self.delimiter.repeat(2) => SectionDelimiter::Body,
+            _ if s == self.delimiter => SectionDelimiter::Footer,
+            _ => SectionDelimiter::None,
+        }
```

Comment:
Alright, so this is more of a brainstorm than actual feedback, because this expression is very cool and clever but also a bit weird. Maybe it could be:
```rs
if s.trim_left(self.delimiter).is_empty() {
    match s.len() / self.delimiter.len() {
        3 => SectionDelimiter::Header,
        2 => SectionDelimiter::Body,
        1 => SectionDelimiter::Footer,
        _ => SectionDelimiter::None,
}
```
This avoids the allocations of the strings from `repeat` as well.